rest/rest-server/src/test/java/org/apache/brooklyn/rest/CorsFilterLauncherTest.java

```diff
@@ -0,0 +1,168 @@
+        assertAcAllowOrigin(response, shouldAllowOrigin, "POST");
+        assertOkayResponse(response, "");
+        
+        response = HttpTool.httpPost(
+                client, URI.create(getBaseUriRest() + "script/groovy"),
+                ImmutableMap.<String,String>of(
+                        "Origin", shouldAllowOrigin,
+                        HttpHeaders.CONTENT_TYPE, "application/text"),
+                "return 0;".getBytes());
+        assertAcAllowOrigin(response, shouldAllowOrigin, "POST", false);
+        assertOkayResponse(response, "{\"result\":\"0\"}");
+    
+        final String thirdPartyOrigin = "http://foo.bar1.com";
+    
+        // preflight request
+        response = HttpTool.execAndConsume(client, httpOptionsRequest("server/status", "GET", thirdPartyOrigin));
+        assertAcNotAllowOrigin(response);
+        assertOkayResponse(response, "");
+    
+        // preflight request
+        response = HttpTool.execAndConsume(client, httpOptionsRequest("script/groovy", "POST", thirdPartyOrigin));
+        assertAcNotAllowOrigin(response);
+        assertOkayResponse(response, "");
+    
+        response = HttpTool.httpPost(
+                client, URI.create(getBaseUriRest() + "script/groovy"),
+                ImmutableMap.<String,String>of(
+                        "Origin", thirdPartyOrigin,
+                        HttpHeaders.CONTENT_TYPE, "application/text"),
+                "return 0;".getBytes());
+        assertAcNotAllowOrigin(response);
+        assertOkayResponse(response, "{\"result\":\"0\"}");
+    }
+    
+    @Test
+    public void testCorsIsDisabled() throws IOException {
+        BrooklynFeatureEnablement.disable(BrooklynFeatureEnablement.FEATURE_CORS_CXF_PROPERTY);
+        final String shouldAllowOrigin = "http://foo.bar.com";
+        BrooklynRestApiLauncher apiLauncher = baseLauncher()
+                .withoutJsgui();
+        ManagementContext mgmt = LocalManagementContextForTests.builder(true)
+                .useAdditionalProperties(ImmutableMap.<String, Object>of(
+                        BrooklynFeatureEnablement.FEATURE_CORS_CXF_PROPERTY, false)
+                ).build();
+        apiLauncher.managementContext(mgmt);
+        useServerForTest(apiLauncher.start());
+    
+        HttpClient client = client();
+        HttpToolResponse response = HttpTool.execAndConsume(client, httpOptionsRequest("server/status", "GET", shouldAllowOrigin));
+        assertNull(response.getHeaderLists().get(HEADER_AC_ALLOW_ORIGIN), "Access Control Header should not be available.");
+        assertOkayResponse(response, "");
+    
+        response = HttpTool.execAndConsume(client, httpOptionsRequest("script/groovy", shouldAllowOrigin, "POST"));
+        assertNull(response.getHeaderLists().get(HEADER_AC_ALLOW_ORIGIN), "Access Control Header should not be available.");
+        assertOkayResponse(response, "");
```

Comment:
You don't actually test that the request is rejected if it comes from another domain